dev/Installer/src/install.cpp

```diff
@@ -0,0 +1,172 @@
+    const BYTE* data = reinterpret_cast<BYTE*>(::LockResource(hresource));
+    THROW_LAST_ERROR_IF_NULL(data);
+    DWORD size = ::SizeofResource(hmodule, hrsrc);
+    wil::com_ptr<IStream> stream = ::SHCreateMemStream(data, size);
+    return stream;
+}
+
+std::unique_ptr<PackageProperties> get_package_properites_from_stream(wil::com_ptr<IStream>& stream)
+{
+    // Get PackageId from the manifest.
+    auto factory = wil::CoCreateInstance<AppxFactory, IAppxFactory>();
+    wil::com_ptr<IAppxPackageReader> reader;
+    THROW_IF_FAILED(factory->CreatePackageReader(stream.get(), wil::out_param(reader)));
+    wil::com_ptr<IAppxManifestReader> manifest;
+    THROW_IF_FAILED(reader->GetManifest(wil::out_param(manifest)));
+    wil::com_ptr<IAppxManifestPackageId> id;
+    THROW_IF_FAILED(manifest->GetPackageId(&id));
+
+    // Populate properties from the manifest PackageId
+    auto properties = std::make_unique<PackageProperties>();
+    THROW_IF_FAILED(id->GetPackageFullName(&properties->full_name));
+    THROW_IF_FAILED(id->GetPackageFamilyName(&properties->family_name));
+    APPX_PACKAGE_ARCHITECTURE arch;
+    THROW_IF_FAILED(id->GetArchitecture(&arch));
+    properties->architecture = static_cast<ProcessorArchitecture>(arch);
+    THROW_IF_FAILED(id->GetVersion(&properties->version));
+
+    return properties;
+}
+
+wil::com_ptr<IStream> open_temp_file(_In_ PCWSTR path)
+{
+    wil::com_ptr<IStream> outstream;
+    THROW_IF_FAILED(SHCreateStreamOnFileEx(path, STGM_WRITE | STGM_READ | STGM_SHARE_DENY_WRITE | STGM_CREATE, FILE_ATTRIBUTE_NORMAL, TRUE, nullptr, wil::out_param(outstream)));
+    return outstream;
+}
+
+winrt::hresult deploy_package_from_resource(ResourcePackageInfo resource, bool quiet) noexcept
+{
+    // Get package properties by loading the resource as a stream and reading the manifest.
+    wil::com_ptr<IStream> pkgstream;
+    std::unique_ptr<PackageProperties> properties;
+    try
+    {
+        pkgstream = get_resource_stream(resource.Id, resource.ResourceType);
+        properties = get_package_properites_from_stream(pkgstream);
+    }
+    catch (wil::ResultException ex)
+    {
+        return ex.GetErrorCode();
+    }
+
+    // Skip non-applicable architectures.
+    if (!is_architecture_applicable(properties->architecture))
+    {
+        return S_OK;
+    }
+
+
+    wchar_t package_filename[MAX_PATH];
+    RETURN_LAST_ERROR_IF(0 == GetTempFileName(std::filesystem::temp_directory_path().c_str(), L"PRP", 0u, package_filename));
+
+    if (!quiet)
+    {
+        printf("Package Full Name: %ls\n", properties->full_name.get());
+        printf("Temp package path: %ls\n", package_filename);
+    }
+
+    // Write the package to a temp file. The PackageManager APIs require a Uri.
+    wil::com_ptr<IStream> outstream;
+    try
+    {
+        outstream = open_temp_file(package_filename);
+    }
+    catch (wil::ResultException ex)
+    {
+        return ex.GetErrorCode();
+    }
+ 
+    ULARGE_INTEGER streamSize = { 0 };
```

Comment:
Fixed the nit.  The GetTempFileName() call creates a unique file because that is specified in the arguments and then returns the name of that unique file. Unless that method is lying and not actually creating a unique filename I think this API should be sufficient for these purposes.